proxy/http2/Http2Stream.cc

```diff
@@ -241,17 +243,22 @@ Http2Stream::do_io_close(int /* flags */)
 {
   current_reader = NULL; // SM on the way out
   if (!sent_delete) {
+    SCOPED_MUTEX_LOCK(lock, this->mutex, this_ethread());
```

Comment:
The reenable is concerning.  Agreed.  That should happen very rarely, but would be a bear when it happens.  I was mostly concerned with holding the lock while doing state changes to the Stream object (changing closed, sent_delete and parent).  I'll restructure to drop the lock before going into reenable.